src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -707,6 +709,23 @@ Copyright (c) .NET Foundation. All rights reserved.
       </_RestoreGraphEntry>
     </ItemGroup>
 
+    <!-- Use packages.config -->
+    <ItemGroup Condition=" '$(RestoreProjectStyle)' == 'PackagesConfig' ">
+      <_RestoreGraphEntry Include="$([System.Guid]::NewGuid())">
+        <Type>ProjectSpec</Type>
+        <ProjectUniqueName>$(MSBuildProjectFullPath)</ProjectUniqueName>
+        <ProjectPath>$(MSBuildProjectFullPath)</ProjectPath>
+        <ProjectName>$(_RestoreProjectName)</ProjectName>
+        <ProjectStyle>$(RestoreProjectStyle)</ProjectStyle>
+        <TargetFrameworks>@(_RestoreTargetFrameworksOutputFiltered)</TargetFrameworks>
+        <PackagesConfigPath Condition="Exists('$(MSBuildProjectDirectory)\packages.$(MSBuildProjectName).config')">$(MSBuildProjectDirectory)\packages.$(MSBuildProjectName).config</PackagesConfigPath>
+        <PackagesConfigPath Condition="Exists('$(MSBuildProjectDirectory)\packages.config')">$(MSBuildProjectDirectory)\packages.config</PackagesConfigPath>
+        <RestorePackagesWithLockFile>$(RestorePackagesWithLockFile)</RestorePackagesWithLockFile>
+        <NuGetLockFilePath>$(NuGetLockFilePath)</NuGetLockFilePath>
```

Comment:
Do you want/need to enable this? 
I had been under the impression this is not a scenario we want to support is it?

Reply:
we support it for PR projects, so why not. If someone sets it in their `Directory.Build.props`, it makes sense for it to apply to all projects, not just some of them.

Reply:
We don't really support custom locations for the packages.config file, why would we for the lock file?

Reply:
Just because some things work differently in PC vs PR, I don't think it's a good idea to intentionally make everything work differently. And just because one feature has a limitation doesn't mean we must build the same limitation into other features.

Reply:
I disagree with that sentiment. 

Features in one management style should work in a way that makes sense in that specific management style.
We shouldn't try to make packages.config and PackageReference similar in a peripheral manner. 
I don't expect anyone would ask for this feature, so why add it. 

Side note, will this even work in VS? I don't see it being read anywhere, but maybe I missed it.

Reply:
I agree that if having to decide between making a feature work well in each project style, or working less well just to make it consistent, it's better to change it to work well, even if there's some differences of the feature across the different project styles.

However, in this case it feels like removing a feature because a different feature for the same project style doesn't have an equivalent functionality. Removing it doesn't make anything better for customers. We have to support it anyway for PR, so I don't think it adds technical debt.

Project metadata is read in `SolutionRestoreJob` in VS, here's the specific line for reading `NuGetLockFilePath`: https://github.com/NuGet/NuGet.Client/blob/0ec447be20273f5e9ae755b3309932f5005162fd/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L531

Reply:
> However, in this case it feels like removing a feature because a different feature for the same project style doesn't have an equivalent functionality

I would again argue that the differences between packagereference and packages.config are so vast that they are effectively different products. 

It doesn't add technical debt, I just don't thing it adds anything :) 

Anyway don't want to be blocking this PR on this.